### PR TITLE
Refresh UI for full iPad support

### PR DESCRIPTION
**Pull request by nCastle1**
This PR updates indoor routing to have a new UI, inspired by modern consumer maps apps. 

There are a lot of changes to support that, including adding a more binding-style approach to keeping the UI updated. 

There are a few things that could probably be improved. A few things I'd like to discuss with reviewers:
* I'm wondering if instead of hiding and showing cards in the bottom/side sheet, it would be better to  use a navigation controller
* I expose a UI state property for the overall states (e.g. searching for route, route found) and the UI reacts to changes there. I'm wondering if it would be better to expose _all_ UI properties directly and update those when transitioning state, rather than leaving it up to the view to decide.
* I'm currently not animating anything in the UI. I wonder if that's OK, or if it would be worth the time to add transition animations
* Each individual card (e.g. location card, route card, search card) is implemented as a UIView. I'm wondering if it would be better to have those be child UIViewControllers instead?

The MVVM changes get this much closer to being something that could be unit tested (and there's enough going on that testing might be worthwhile), but there are still a few things that are tightly coupled. I'd like to in the future think about avoiding static global state and maybe use something like dependency injection/passing dependencies around. I think that would probably be needed to support multi-window workflows as well.

### CODE REVIEW

File: IndoorRouting/Models/Room.cs [hunk 1]
```diff
@@ -0,0 +1,217 @@
...
+                // Get room attribute from the settings. First attribute should be set as the searchable one
+                var roomAttribute = AppSettings.CurrentSettings.ContactCardDisplayFields[0];
+                var roomNumber = feature.Attributes[roomAttribute];
+
+                var employeeNameLabel = string.Empty;
+                if (AppSettings.CurrentSettings.ContactCardDisplayFields.Count > 1)
+                {
+                    var employeeNameAttribute = AppSettings.CurrentSettings.ContactCardDisplayFields[1];
+                    var employeeName = feature.Attributes[employeeNameAttribute];
+                    employeeNameLabel = employeeName as string ?? string.Empty;
```
Comment (ZackAllen) [inline IndoorRouting/Models/Room.cs:149]: I like this pattern. Its pretty slick.

File: iOS/ApplicationTheme.cs [hunk 2]
```diff
@@ -0,0 +1,68 @@
...
+        public static UIColor ActionBackgroundColor = new UIColor(0.071f, 0.475f, 0.757f, 1.00f);
+        public static UIColor ActionForegroundColor = UIColor.White;
+        public static UIColor SelectionBackgroundColor = ActionBackgroundColor;
+        public static UIColor SelectionForegroundColor = ActionForegroundColor;
+        // Accessory button is a light/dark responsive color defined in the asset catalog
+        public static UIColor AccessoryButtonColor = UIColor.FromName("AccessoryButtonColor");
+        public static nint ActionButtonHeight = 44;
+        public static UIFont HeaderFont = UIFont.BoldSystemFontOfSize(28);
+    }
+    /*
```
Comment (ZackAllen) [inline iOS/ApplicationTheme.cs:45]: Maybe you should add a description of this alternate theme.

File: iOS/Models/SettingsTableSource.cs [hunk 3]
```diff
@@ -0,0 +1,135 @@
...
+        public SettingsTableSource(MapViewModel viewModel) => _viewModel = viewModel;
+
+        /// <summary>
+        /// Returns the setting cell for a particular row
+        /// </summary>
+        public override UITableViewCell GetCell(UITableView tableView, NSIndexPath indexPath)
+        {
+            switch (indexPath.Row)
+            {
+                case 0:
```
Comment (ZackAllen) [inline iOS/Models/SettingsTableSource.cs:36]: I would consider adding comments describing the cases. We dont have context to what they are here.

File: iOS/Views/MapViewCards/LocationInfoCard.cs [hunk 4]
```diff
@@ -0,0 +1,148 @@
...
+            };
+
+            AddSubviews(_startDirectionsButton, closeButton, _primaryLabel, _secondaryLabel);
+
+            NSLayoutConstraint.ActivateConstraints(new[]
+            {
+                _primaryLabel.LeadingAnchor.ConstraintEqualTo(LeadingAnchor, ApplicationTheme.Margin),
+                _primaryLabel.TopAnchor.ConstraintEqualTo(TopAnchor, ApplicationTheme.Margin),
+                _primaryLabel.TrailingAnchor.ConstraintEqualTo(closeButton.LeadingAnchor, -ApplicationTheme.Margin),
+                //
```
Comment (ZackAllen) [inline iOS/Views/MapViewCards/LocationInfoCard.cs:69]: Not sure if these empty comments fit the standard format.

File: iOS/Views/MapViewCards/RouteResultCard.cs [hunk 5]
```diff
@@ -0,0 +1,164 @@
...
+            StringBuilder walkTimeStringBuilder = new StringBuilder();
+
+            // Add walk time and distance label
+            if (firstRoute.TotalTime.Hours > 0)
+            {
+                walkTimeStringBuilder.Append($"{firstRoute.TotalTime.Hours} h {firstRoute.TotalTime.Minutes} m");
+            }
+            else
+            {
+                walkTimeStringBuilder.Append($"{firstRoute.TotalTime.Minutes + 1} min");
```
Comment (ZackAllen) [inline iOS/Views/MapViewCards/RouteResultCard.cs:135]: Is there a convenient way to localize "min" here? Is that localization handled elsewhere?

**Comment by mhdostal** [comment]
@nCastle1 Aside from the Slack conversation we had (about the BottomSheetVC handling the gestures and subsequent sizing, but having the owning VC do the initial placement), I had the following question for @esreli, regarding the `UIAppearance` API in iOS and whether that would be appropriate to use in place of some of the "Application Theme" class:

There is an “Application Theme” class (https://github.com/Esri/indoor-routing-xamarin/pull/61/files#diff-1a523bd9fa0dbf998008b37579210e12R399) used to set colors and other UI-related stuff.  I haven’t used the UIAppearance API, but would that be appropriate to use in this situation (at least for some of the properties)?

Otherwise, I've gone through everything and it looks good.

**Comment by esreli** [comment]
re: https://github.com/Esri/indoor-routing-xamarin/pull/61#issuecomment-629494845

I do think `UIAppearance` presents an opportunity to globally skin UIKit components. From the looks of it, not all ApplicationTheme constants would interface with `UIAppearance`, though some seem as though they could.

@nCastle1 have you considered/ what sorts of considerations have you made surrounding interfacing with `ApplicationTheme` with `UIAppearance`? Would you consider this an Xamarin.iOS best practice?

**Comment by nCastle1** [comment]
@esreli @mhdostal The reason I went with a separate class with constant values was because there were several properties I couldn't set through UIAppearance (primarily margins in layout constraints) and I didn't want to make the developer go to two places to customize things. Because the use of out-of-the-box UIKit views is minimal (most things are custom views), so there was limited value in UIAppearance.

Using `UIAppearance` would demonstrate more of the native UIKit APIs; I don't know how we want to prioritize that. If that is a priority, I can update this PR to use UIAppearance where possible.

**Comment by esreli** [comment]
https://github.com/Esri/indoor-routing-xamarin/pull/61#issuecomment-630473298

That's reasonable and i'm fine with how you have implemented UI constants.

**Comment by mhdostal** [comment]
@NathanCastle Summarizing the bottom sheet discussion we had...

Typically, when displaying a child ViewController, the displaying VC does the initial placement of the child view and subsequent changes to the child display is handled by the child VC.  So the gesture recognition and adjusting of constraints to get the small, medium and large sizes would be handled by the child.

**Comment by mhdostal** [comment]
@NathanCastle Looks really good.  I think this approach is much better than previously.  Had a question bout the `TouchTransparentView`, why is this necessary?  It looks like you're overriding the provided `view` property of the base `UIViewController`, which is unusual.

File: iOS/Controllers/BottomSheetViewController.cs [hunk 6]
```diff
@@ -34,184 +33,200 @@ public sealed class BottomSheetViewController : UIViewController
...
+                        return View.Superview.Frame.Height + ApplicationTheme.Margin - View.Superview.SafeAreaInsets.Top - (2 * ApplicationTheme.Margin);
+                    case UIUserInterfaceSizeClass.Regular:
+                    default:
+                        return View.Superview.Frame.Height - View.Superview.SafeAreaInsets.Top - (2 * ApplicationTheme.Margin) - View.Superview.SafeAreaInsets.Bottom - (2 * ApplicationTheme.Margin);
+                }
+            }
+        }
 
-            blurView.ContentView.AddSubview(DisplayedContentView);
+        public override void ViewWillAppear(bool animated)
```
Comment (mhdostal) [inline iOS/Controllers/BottomSheetViewController.cs:110]: You have call `super.viewWillAppear(animated)` at some point.  I'd do it right at the top.  Same goes for all the `view.` methods you override.

https://developer.apple.com/documentation/uikit/uiviewcontroller/1621510-viewwillappear?language=swift

"If you override this method, you must call super at some point in your implementation."

**Pull request merged**